eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -154,38 +160,68 @@ jobs:
     parameters:
       SubscriptionConfiguration: ${{ parameters.CloudConfig.SubscriptionConfiguration }}
       SubscriptionConfigurations: ${{ parameters.CloudConfig.SubscriptionConfigurations }}
+      SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
+      EnvVars: ${{ parameters.EnvVars }}
 
   - template: /eng/common/TestResources/deploy-test-resources.yml
     parameters:
       ServiceDirectory: ${{ parameters.ServiceDirectory }}
       Location: ${{ coalesce(parameters.Location, parameters.CloudConfig.Location) }}
       SubscriptionConfiguration: $(SubscriptionConfiguration)
+      UseFederatedAuth: ${{ parameters.UseFederatedAuth }}
+      EnvVars: ${{ parameters.EnvVars }}
+      ServiceConnection: ${{ parameters.CloudConfig.ServiceConnection }}
 
   - template: /eng/common/testproxy/test-proxy-tool.yml
     parameters:
       runProxy: false
 
   - ${{ parameters.PreTestSteps }}
 
-    # For non multi-config generator use the same build configuration to run tests
-    # We don't need to set it to invoke ctest
-    # Visual Studio generator used in CI is a multi-config generator.
-    # As such, it requires the configuration argument for building and invoking ctest
-  - bash: |
-      export AZURE_CLIENT_ID=$(${{parameters.ServiceDirectory}}_CLIENT_ID)
-      export AZURE_TENANT_ID=$(${{parameters.ServiceDirectory}}_TENANT_ID)
-      export AZURE_CLIENT_SECRET=$(${{parameters.ServiceDirectory}}_CLIENT_SECRET)
+  - ${{ if parameters.UseFederatedAuth }}:
+    - task: AzurePowerShell@5
+      displayName: ctest
+      # Runs only if test-resources are happily deployed.
+      # unit-tests runs for those configs where samples are not ran.
+      # This enables to run tests and samples at the same time as different matrix configuration.
+      # Then unit-tests runs, samples should not run.
```

Comment:
Some tests still need `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, and `AZURE_CLIENT_SECRET` env vars set, when the Live pipelines are run.

Otherwise, they'll fail.
For example:
https://github.com/Azure/azure-sdk-for-cpp/blob/55c241902d6897f9c8b9aee2668c76c5d0723ba4/sdk/identity/azure-identity/test/ut/token_credential_test.cpp#L56

Reply:
Summarizing offline conversation options: 

* Disable client secret tests, open issue, re-enable when resources for client secret tests are available
* Only run tests in playback mode, open issue to re-enable when resources for live client secret tests are available
* Keep the failures as known issues in live testing